### PR TITLE
Adding a DemoRoom sample for Windows Mixed Reality APIs

### PR DESCRIPTION
**Pull request by jbuschdev-msft**
This is a sample to demonstrate best practices with a native C++ DirectX application. 

It's called "DemoRoom" and has a simple environment with a few tools you can use with the Windows Mixed Reality Motion Controllers. It relies on a few other libs that were already put into the MixedRealityToolKit for the "RenderController" sample.

image

### CODE REVIEW

File: SpatialInput/Samples/DemoRoom/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Painting Tool
+  - How to use SourceUpdated events for more stable painting
+      - [SpatialInteractionManager::SourceUpdated](https://docs.microsoft.com/uwp/api/windows.ui.input.spatial.spatialinteractionmanager)
+  - How to paint using paint brush tip (3d model)
+  - How to use touchpad to implement a radio menu
+    - [SpatialInteractionSourceState::ControllerProperties::Touchpad[X/Y]](https://docs.microsoft.com/uwp/api/windows.ui.input.spatial.spatialinteractionsourcestate)
+  - How to use SpatialPointerInteractionSourcePose to display a ray from controller
+    - [SpatialInteractionSourceProperties::SourcePointerPose](https://docs.microsoft.com/uwp/api/windows.ui.input.spatial.spatialinteractionsourcelocation)
+
+Any details around Motion Controller rendering are better explained in the [Render Controller sample](..\RenderController\README.md)
```
Comment (jbuschdev-msft) [inline SpatialInput/Samples/DemoRoom/README.md]: This link is broken, not sure how to do relative paths w/ github.

File: SpatialInput/Libs/Neso/Common/CameraResources.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
+#include "pch.h"
```
Comment (NeerajW) [inline SpatialInput/Libs/Neso/Common/CameraResources.cpp]: Please add Microsoft copyright headers to all files.

File: SpatialInput/Libs/Neso/Engine/EntityStore.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <Neso\Engine\Engine.h>
+#include <Neso\Engine\ComponentStore.h>
+
+namespace Neso
+{
+    class EntityStore : public System<EntityStore>
```
Comment (NeerajW) [inline SpatialInput/Libs/Neso/Engine/EntityStore.h]: All classes are missing class descriptions. Some of these can get confusing with API names.

File: SpatialInput/Libs/Neso/Neso.vcxproj [hunk 4]
```diff
@@ -0,0 +1,234 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (NeerajW) [inline SpatialInput/Libs/Neso/Neso.vcxproj]: What is Neso?

File: SpatialInput/README.md [hunk 5]
```diff
@@ -2,10 +2,12 @@
...
 ## Samples\
 
+* [Demo Room](Samples/DemoRoom) - DirectX 11 UWP C++/WinRT sample application demonstrating a variety of application/uses of Windows Perception APIs. This sample uses the Pbr, Gltf, Neso, and SpatialInputUtilities libraries.
 * [Motion Controller Rendering Sample](Samples/RenderController) - DirectX 11 UWP C++/WinRT sample application demonstrating how to render a motion controller. This sample uses the Pbr, Gltf and SpatialInputUtilities libraries.
 
 ## Libs\
 
-* [Pbr](Libs/Pbr) - Shaders and code for creating and rendering models in DirectX 11 using the PBR Metallic-Roughness model.
 * [Gltf](Libs/Gltf) - Helper functions to parse data returned by TinyGLTF.
+* [Neso](Libs/Neso) - Simple Entity-Component-System (ECS) engine for real-time applications. Depends on Pbr, Gltf, and SpatialInputUtilities libs.
```
Comment (NeerajW) [inline SpatialInput/README.md]: @brycehutchings and @jbuschdev-msft should these samples live in:

https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples ?

File: SpatialInput/Libs/Neso/Engine/EntityStore.h (hunk 3 above)
Comment (jbuschdev-msft) [reply SpatialInput/Libs/Neso/Engine/EntityStore.h]: Done

File: SpatialInput/Libs/Neso/Common/CameraResources.cpp (hunk 2 above)
Comment (jbuschdev-msft) [reply SpatialInput/Libs/Neso/Common/CameraResources.cpp]: Done

File: SpatialInput/Libs/Neso/Neso.vcxproj (hunk 4 above)
Comment (jbuschdev-msft) [reply SpatialInput/Libs/Neso/Neso.vcxproj]: A simple Entity-Component-System library/game engine. The name has no real meaning, just preferred something more unique than "Engine" or "GameEngine" =)

**Pull request merged**